---
title: "Week 07 Question Set"
output: html_document
date: "2025-05-13"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

a)  **What is your current grade in the class? How many issues are open in our research repository?**\
    It looks like I have a 90% if I calculated things correctly? Couple points off assignments - would like to fix that if possible. I have 3 issues open as of today - two seem to be similar issues, ie. related to the final html outputs of a couple past assignments. I can fix those starting tomorrow (Tuesday) - not sure what went wrong with them, so time for some more debug practice!\

b)  **Re-reading this week's reading, what are 2 things that resonate with you now that made little practical sense in week 2. What is one aspect that you still wonder about or is unclear?**\
    Two things that resonate:\

```{=html}
<!-- -->
```
1)  Visualization methods are a lot more fleshed out for me now; over in my world of popgen we do absolutely look at our sequences a lot (I've had to look at mine more than I ever expected to these past few weeks due a program/package issue), but the exploration we did in class helped me become familiar with other methods I hadn't previously used, like IGV.\
2)  The process of using epigenetic and/or transcriptome data is less scary! There's a lot of overlap with regular ol' genomics work, of course, but for some reason my mind partitioned it away as a bit more complicated and confusing. Going through the process myself and then going back to read this paper made me realize that I am now more comfortable working with that kind of data, even if it's still very new to me.\

One aspect that I have more curiosity about is enrichment analysis. I have seen it in its various forms over the years (I feel like most often it's using GO terms but I'm also not often seeking out papers on the subject), but I would love to know how it's determined that certain sets of genes/functions are popping up more than random chance or association with other linked terms. I guess that's a fairly loose topic, but it's a topic that interests me!\

c)  **What is epigenetics and how could it relate to your own research project?**\
    While I'm not touching epigenetics at all in my work, my project focuses on rockfish in Puget Sound with the ultimate motivation to understand why these populations aren't recovering. Several non-genetic reasons have been pointed to (besides the obvious one I have mentioned before, which is that these pops are not connected to the coastal ones), with some being related to environmental conditions and pollution. With the latter point especially, I think epigenetics work comparing expression in Puget Sound rockfish versus their coastal counterparts could be interesting to see if they are under unique stressors that are seen less or are absent in coastal, less polluted waters. That's probably a very reductive example given how vast and complex epigenetics clearly is, but it's the first cool idea that came to mind!!\

d)  **Describe what a GFF file is?**\
    A GFF file, or genome feature file (so I guess saying GFF file is like saying ATM machine), contains information about regions of interest in the genome (i.e. genes for example) and can be used for annotation and downstream analyses, such as differential gene expression (as mentioned in the reading!).\

e)  **What are two ways describing CG motifs could have value?**\
    Two examples of value:

```{=html}
<!-- -->
```
1)  You can look at genomic evolution via the distribution and conservation of CG motifs across the genome, which sounds particularly exciting to me, and\
2)  Looking at epigenetic inheritance, wherein epigenetic changes like methylation are passed onto offspring (which is still a crazy concept, but makes a lot of sense); how could those patterns of inheritance, depending on what regions of the genome they exist in, influence subsequent generations?
