{
  "cells": [
    {
      "cell_type": "raw",
      "metadata": {},
      "source": [
        "---\n",
        "title: \"08-bedtools\"\n",
        "format: html\n",
        "editor: visual\n",
        "---"
      ],
      "id": "455d2c4f"
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "Grabbing the data (again!)\n",
        "\n",
        "\n",
        "```{bash, eval = FALSE}\n",
        "\n",
        "cd ~/home/shared/8TB_HDD_02/lizboggs/bedtools_assign\n",
        "curl -O https://gannet.fish.washington.edu/seashell/bu-mox/scrubbed/120321-cvBS/19F_R1_val_1_bismark_bt2_pe.deduplicated.sorted.bam \n",
        "curl -O https://gannet.fish.washington.edu/seashell/bu-mox/scrubbed/120321-cvBS/19F_R1_val_1_bismark_bt2_pe.deduplicated.sorted.bam.bai\n",
        "\n",
        "```\n",
        "\n",
        "\n",
        "BED file with gene information:\n",
        "\n",
        "\n",
        "```{bash, eval = FALSE}\n",
        "\n",
        "cd ~/home/shared/8TB_HDD_02/lizboggs/bedtools_assign\n",
        "curl -O https://eagle.fish.washington.edu/Cvirg_tracks/C_virginica-3.0_Gnomon_genes.bed\n",
        "\n",
        "```\n",
        "\n",
        "\n",
        "Converting bam to bed:\n",
        "\n",
        "\n",
        "```{bash, eval = FALSE}\n",
        "\n",
        "/home/shared/bedtools2/bin/bedtools bamtobed \\\n",
        "-i ~/lizboggs/bedtools_assign/19F_R1_val_1_bismark_bt2_pe.deduplicated.sorted.bam \\\n",
        "> ~/lizboggs/bedtools_assign/08-19F.bed\n",
        "\n",
        "```\n",
        "\n",
        "\n",
        "Running BEDtools `coverage`\n",
        "\n",
        "\n",
        "```{bash, eval = FALSE}\n",
        "\n",
        "/home/shared/bedtools2/bin/bedtools coverage \\\n",
        "-a ~/lizboggs/bedtools_assign/C_virginica-3.0_Gnomon_genes.bed \\\n",
        "-b ~/lizboggs/bedtools_assign/08-19F.bed \\\n",
        "> ~/lizboggs/bedtools_assign/08-gene-19F-coverage.out\n",
        "\n",
        "```\n",
        "\n",
        "\n",
        "Getting a bed file of transposable elements and lncRNAs:\n",
        "\n",
        "\n",
        "```{bash, eval = FALSE}\n",
        "\n",
        "cd ~/home/shared/8TB_HDD_02/lizboggs/bedtools_assign\n",
        "curl -O http://owl.fish.washington.edu/halfshell/genomic-databank/cgigas_uk_roslin_v1_gene.gff\n",
        "curl -O http://owl.fish.washington.edu/halfshell/genomic-databank/cgigas_uk_roslin_v1_rm.te.bed\n",
        "curl -O http://owl.fish.washington.edu/halfshell/genomic-databank/cgigas_uk_roslin_v1_lncRNA.gff\n",
        "\n",
        "```\n",
        "\n",
        "\n",
        "Running BEDtools `intersect` to find where exactly A and B match up with each other\n",
        "\n",
        "\n",
        "```{bash, eval = FALSE}\n",
        "\n",
        "/home/shared/bedtools2/bin/bedtools intersect \\\n",
        "-a ~/lizboggs/bedtools_assign/cgigas_uk_roslin_v1_gene.gff \\\n",
        "-b ~/lizboggs/bedtools_assign/cgigas_uk_roslin_v1_rm.te.bed \\\n",
        "> ~/lizboggs/bedtools_assign/08-gene-TE-intersect.out\n",
        "\n",
        "head -2 ~/lizboggs/bedtools_assign/08-gene-TE-intersect.out\n",
        "\n",
        "```\n",
        "\n",
        "\n",
        "Running BEDtools `closest` to find the nearest features between A and B\n",
        "\n",
        "\n",
        "```{bash, eval = FALSE}\n",
        "\n",
        "/home/shared/bedtools2/bin/bedtools closest \\\n",
        "-a ~/lizboggs/bedtools_assign/cgigas_uk_roslin_v1_lncRNA.gff\\\n",
        "-b ~/lizboggs/bedtools_assign/cgigas_uk_roslin_v1_gene.gff \\\n",
        "> ~/lizboggs/bedtools_assign/08-lnc-gene-closet.out\n",
        "\n",
        "```"
      ],
      "id": "ece1469f"
    }
  ],
  "metadata": {
    "kernelspec": {
      "display_name": "Python 3",
      "language": "python",
      "name": "python3"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 5
}